### PR TITLE
raftstore: add hibernate-timeout to prevent leader become hibernated too fast  (#7955)

### LINKED ISSUE #7747: Fail to evict leader when rolling update a new hibernated cluster
## Bug Report

If there are too many leaders or transfer leader just goes too slow, rolling update a cluster can take too much time. If the cluster enables hibernate region at the time, it's possible that all updated nodes will not have any leader because updated leader becomes hibernated very quickly and outdated follower will start election when lost heartbeat from leader for a long time. So it becomes slower because more leaders need to be evicted during rolling update.

A possible solution is to add a hibernate timeout configuration. Leader will not go to sleep before the timeout. So it's safe as long as the hibernate timeout is larger than the update time. And the timeout can be set to a smaller value after the cluster has enabled hibernate region on all nodes.

Another solution is make the configuration be changed online. So users can enable/disable hibernate regions whenever they feel safe.

We need to fix the issue in 3.0.x, 3.1.x and 4.0.

### PR DESCRIPTION
**Pull request by ti-srebot**
cherry-pick #7955 to release-3.0

---

Signed-off-by: linning <linningde25@gmail.com>

### What problem does this PR solve?

Issue Number: close #7747 

Problem Summary:

Rolling update to hibernated cluster may take too long, because the hibernated leader will not send heartbeat to non-hibernated followers and cause them to start election.

### What is changed and how it works?

What's Changed:

Add `hibernate-timeout` config to raftstore, new elected leader will not be hibernated within this timeout, and this config can be changed online when the rolling update is finished.

### Release note 
- add hibernate-timeout config to prevent leader become hibernated too fast

### CODE REVIEW

File: components/test_raftstore/src/common-test.toml [hunk 1]
```diff
@@ -0,0 +1,108 @@
+[readpool]
```
Comment (BusyJay) [inline components/test_raftstore/src/common-test.toml:1]: Why add the file?

Comment (NingLin-P) [reply components/test_raftstore/src/common-test.toml]: It is a mistake, the file removed.

**Pull request merged**